recipes/cppserver/all/patches/00001-cmake-clean-up.patch

```diff
@@ -0,0 +1,141 @@
+@@ -18,50 +18,50 @@ if(DOXYGEN_FOUND)
+ endif()
+ 
+ # CMake module path
+-set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
++# set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
+ 
+ # Compiler features
+-include(SetCompilerFeatures)
+-include(SetCompilerWarnings)
+-include(SetPlatformFeatures)
+-include(SystemInformation)
++# include(SetCompilerFeatures)
++# include(SetCompilerWarnings)
++# include(SetPlatformFeatures)
++# include(SystemInformation)
+ 
+ # External packages
+-if(CYGWIN)
+-  set(OPENSSL_ROOT_DIR "/usr/lib")
+-  set(OPENSSL_INCLUDE_DIR "/usr/include")
+-  set(OPENSSL_CRYPTO_LIBRARY "/usr/lib/libcrypto.dll.a")
+-  set(OPENSSL_SSL_LIBRARY "/usr/lib/libssl.dll.a")
+-elseif(MINGW)
+-  set(OPENSSL_ROOT_DIR "${CMAKE_CURRENT_SOURCE_DIR}/modules/OpenSSL/MinGW")
+-  set(OPENSSL_USE_STATIC_LIBS TRUE)
+-elseif(MSVC)
+-  set(OPENSSL_ROOT_DIR "${CMAKE_CURRENT_SOURCE_DIR}/modules/OpenSSL/VS")
+-  set(OPENSSL_USE_STATIC_LIBS TRUE)
+-  set(OPENSSL_MSVC_STATIC_RT TRUE)
+-endif()
++# if(CYGWIN)
++#   set(OPENSSL_ROOT_DIR "/usr/lib")
++#   set(OPENSSL_INCLUDE_DIR "/usr/include")
++#   set(OPENSSL_CRYPTO_LIBRARY "/usr/lib/libcrypto.dll.a")
++#   set(OPENSSL_SSL_LIBRARY "/usr/lib/libssl.dll.a")
++# elseif(MINGW)
++#   set(OPENSSL_ROOT_DIR "${CMAKE_CURRENT_SOURCE_DIR}/modules/OpenSSL/MinGW")
++#   set(OPENSSL_USE_STATIC_LIBS TRUE)
++# elseif(MSVC)
++#   set(OPENSSL_ROOT_DIR "${CMAKE_CURRENT_SOURCE_DIR}/modules/OpenSSL/VS")
++#   set(OPENSSL_USE_STATIC_LIBS TRUE)
++#   set(OPENSSL_MSVC_STATIC_RT TRUE)
++# endif()
+ find_package(OpenSSL REQUIRED)
+-if(WIN32)
++if(FALSE)
+   find_package(Crypt)
+   find_package(WinSock)
+ endif()
+ 
+ # Modules
+-add_subdirectory("modules")
++# add_subdirectory("modules")
+ 
+ # Link libraries
+-list(APPEND LINKLIBS ${OPENSSL_LIBRARIES})
++list(APPEND LINKLIBS CONAN_PKG::openssl)
+ if(WIN32)
+-  list(APPEND LINKLIBS ${CRYPT_LIBRARIES})
+-  list(APPEND LINKLIBS ${WINSOCK_LIBRARIES})
++  list(APPEND LINKLIBS crypt32)
++  list(APPEND LINKLIBS mswsock ws2_32)
+ endif()
+-list(APPEND LINKLIBS cppcommon)
++list(APPEND LINKLIBS CONAN_PKG::cppcommon)
+ 
+ # OpenSSL libraries
+ message(STATUS "OpenSSL version: ${OPENSSL_VERSION} ${OPENSSL_INCLUDE_DIR} ${OPENSSL_LIBRARIES}")
+ 
+ # System directories
+-include_directories(SYSTEM "${CMAKE_CURRENT_SOURCE_DIR}/modules")
++# include_directories(SYSTEM "${CMAKE_CURRENT_SOURCE_DIR}/modules")
+ 
+ # Library
+ file(GLOB_RECURSE LIB_HEADER_FILES "include/*.h")
+@@ -70,14 +70,15 @@ file(GLOB_RECURSE LIB_SOURCE_FILES "source/*.cpp")
+ add_library(cppserver ${LIB_HEADER_FILES} ${LIB_INLINE_FILES} ${LIB_SOURCE_FILES})
+ set_target_properties(cppserver PROPERTIES COMPILE_FLAGS "${PEDANTIC_COMPILE_FLAGS}" FOLDER "libraries")
+ target_include_directories(cppserver PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}/include")
+-target_link_libraries(cppserver ${LINKLIBS} asio)
++target_link_libraries(cppserver ${LINKLIBS} CONAN_PKG::asio)
++target_compile_features(cppserver PUBLIC cxx_std_17)
```

Comment:
this seems out of place 🤔

Reply:
You mean the `cxx_std_17` feature, right?

This is similar to #4001.

This is one of those cases where the author handles this upstream with a CMake module. The relevant parts are:

- `CppServer/.gitlinks`:
```
# Scripts
build build https://github.com/chronoxor/CppBuildScripts.git master
cmake cmake https://github.com/chronoxor/CppCMakeScripts.git master
```
https://github.com/chronoxor/CppServer/blob/c67a656cac13c86e005f6ffcb288b9ffa8daf132/.gitlinks#L8-L10

- `CppServer/CMakeLists.txt`:
```cmake
# Compiler features
include(SetCompilerFeatures)
include(SetCompilerWarnings)
include(SetPlatformFeatures)
include(SystemInformation)
```
https://github.com/chronoxor/CppServer/blob/c67a656cac13c86e005f6ffcb288b9ffa8daf132/CMakeLists.txt#L23-L27

- `CppCMakeScripts/SetCompilerFeatures.cmake`:
```cmake
# Compiler features


# Choose C++ standard
set(CMAKE_CXX_STANDARD 17)
```

https://github.com/chronoxor/CppCMakeScripts/blob/4152297a1c28cfd2ea63ea9a4bdfcb50aa2c05d9/SetCompilerFeatures.cmake#L1-L4

In the other PR I said that this is done through git submodules, but I was mistaken. It's actually handled with a custom dependency management system (the author calls it GIL).